### PR TITLE
Added pyls.extraSysPath. Fixes #381

### LINKED ISSUE #381: Allow passing sys_path to Jedi from the settings
Additionally to `source_roots`, it should be possible to pass additional `sys_path` from the settings. Issues arise when using Python2/Python3 codebases and simply setting `PYTHONPATH`.

### PR DESCRIPTION
**Pull request by Kronuz**
This adds `extraSysPath` to extend `extra_sys_path` passed to Jedi.

### CODE REVIEW

File: pyls/workspace.py [hunk 1]
```diff
@@ -93,9 +94,12 @@ def source_roots(self, document_path):
 
     def _create_document(self, doc_uri, source=None, version=None):
         path = uris.to_fs_path(doc_uri)
+        extra_sys_path = self.source_roots(path)
+        settings = self._config.settings(doc_uri)
+        extra_sys_path.extend(settings.get('extraSysPath', []))
```
Comment (gatesn) [inline pyls/workspace.py:99]: Extend? Or prepend? I wonder if an explicit config override should be put at the front of the sys path

**Review by gatesn — CHANGES_REQUESTED** [review]
Think we should have `settings.get('extraSysPath', []) + self.source_roots(path)`

**Comment by gagbo** [comment]
Should the work on this start again, would this be merged ? I don't understand why it stalled exactly, and this is exactly the feature I'm missing.

I think I prefer prepending the extra_sys_path (or maybe give one after and one before options), because I would want to put specific `PYTHONPATH` values here, which are inserted early in `sys.path()` when running a python program. Especially when, as @gatesn says, it is an override.

**EDIT** : Actually, setting PYTHONPATH correctly should do the trick already (from adding a few logs in 0.26 code). At least, `sys_path` has a correct value, but jedi still doesn't pick up the folder like I want. I'll make a minimal example soon hopefully

**Comment by ccordoba12** [comment]
This was implemented in PR #680.

**Pull request closed without merging**